MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -137,15 +131,75 @@ struct EventDetails {
         
         modifiedEventDictionary["batteryPluggedIn"] = batteryPluggedIn
         modifiedEventDictionary["batteryLevel"] = batteryLevel
-        modifiedEventDictionary["applicationState"] = applicationState
+        modifiedEventDictionary["applicationState"] = applicationState.telemetryString
         
         return modifiedEventDictionary
     }
 }
 
 extension MMEEventsManager {
     func addDefaultEvents(routeController: RouteController) -> [String: Any] {
-        return EventDetails(routeController: routeController, session: routeController.sessionState).convertedToDictionary()
+        return EventDetails(routeController: routeController, session: routeController.sessionState).eventDictionary
+    }
+    
+    func navigationDepartEvent(routeController: RouteController) -> [String: Any] {
```

Comment:
Instead of a series of separate methods that each sets a different `event` value, we should have a single method that takes an `MMEEventType` as a parameter; depending on the passed-in `eventType`, we can add additional items to `eventDictionary` as needed.